DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs

```diff
@@ -82,24 +86,39 @@ public IActionResult AllAdmins()
         [HttpGet]
         public IActionResult EditAdminRoles(int adminId)
         {
+            var centreId = User.GetCentreId();
             var adminUser = userDataService.GetAdminUserById(adminId);
-            if (adminUser == null)
+
+            if (adminUser == null || adminUser.CentreId != centreId)
             {
                 return NotFound();
             }
 
-            var centreId = User.GetCentreId();
             var categories = courseCategoriesDataService.GetCategoriesForCentreAndCentrallyManagedCourses(centreId);
             categories = categories.Prepend(new Category { CategoryName = "All", CourseCategoryID = 0 });
+            var numberOfAdmins = numberOfAdministratorsHelper.GetCentreAdministratorNumbers(centreId);
 
-            var model = new EditRolesViewModel(adminUser, centreId, categories);
+            var model = new EditRolesViewModel(adminUser, centreId, categories, numberOfAdmins);
             return View(model);
         }
 
         [Route("{adminId:int}/EditAdminRoles")]
         [HttpPost]
         public IActionResult EditAdminRoles(EditRolesViewModel model, int adminId)
         {
+            var centreId = User.GetCentreId();
+            var adminUser = userDataService.GetAdminUserById(adminId);
+
+            if (adminUser == null || adminUser.CentreId != centreId)
+            {
+                return NotFound();
+            }
+
+            if (NewUserRolesExceedAvailableSpots(model, adminUser))
+            {
+                return new StatusCodeResult(500);

```

Comment:
Do we want to throw something more explicit here so we can track if this is occurring?

Reply:
At time of writing, we have decided to go with this. I believe Stella is still waiting to hear back if HEE would prefer a custom error page. I've added a logger though as its a good idea.

Reply:
I was more thinking of throwing an exception with a message, which will result in a 500 anyway, but I think an explicit log and return 500 is probably clearer.

Reply:
HEE has confirmed that throwing is OK here - I think Alex is right that it would make sense to just throw a custom error, (which will be handled by default with logging & redirect to 500)